libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/app/MobileSyncSDKManager.java

```diff
@@ -64,9 +64,10 @@ private static void init(Context context, Class<? extends Activity> mainActivity
     		INSTANCE = new MobileSyncSDKManager(context, mainActivity, loginActivity);
     	}
 
+		initInternal(context);
```

Comment:
This call is meant to be after the upgrade step, by design. There's a lot of internal initialization that occurs there, and without running upgrade steps, it would lead to permanent corruption. For example, `SmartStore` DB opening calls. Let's move this back below.

Reply:
I looked at that code path and it didn't look like `initInternal()` did anything that would cause issues across all SDK manager classes.  If it really does need to be after the upgrade steps, we need to break the `HttpAccess.init()` out of `initInternal()`.  Do you foresee that causing issues if we have to revert this reordering?